lib/swaggerUtil.js

```diff
@@ -24,6 +24,106 @@ exports.validateJSONType = function (schema, value) {
     return result;
 };
 
+exports.getObjectsWithDiscriminator = function (object) {
+    var map = {};
+    var properties = object.properties;
+    var discriminator = '';
+    if (_.has(object, 'allOf') || _.has(object, 'items.allOf')) {
+        var inheritedModels = object.allOf ? object.allOf : object.items.allOf;
+        map.isArray = object.type === 'array' ? true : undefined;
+        inheritedModels.forEach(function (inheritedModel) {
+            _.merge(map, exports.getObjectsWithDiscriminator(inheritedModel));
+        });
+    }
+    if (_.has(object, 'discriminator') || _.has(object, 'items.discriminator')) {
```

Comment:
given the specially handling of the array case that happens inside this block, why not simply handle the two cases (arrays and objects) independently at the "top level"?